hnn_core/gui/_viz_manager.py

```diff
@@ -958,6 +983,15 @@ def add_figure(self, b=None):
                 # paint fig in axis
                 self._simulate_edit_figure(fig_name, ax_name, sim_name,
                                            plot_type, {}, "plot")
+            # template post-processing
+            fig_key = self.data['fig_idx']['idx'] - 1
+            _postprocess_template(template_name,
+                                  fig=self.figs[fig_key],
```

Comment:
@ntolley 
This was the PR I was showing you that was glitching and preventing further figure creation after deleting figures a few times. The issue was at this line where I was previously indexing `self.figs` with a the length of itself, because I assumed it was a list and I wanted the latest created figure. Turns out that`self.figs` is a dict with ints as keys and the length of the dict doesn't correspond to the keys after you delete from and add to it.  Using the int stored in `self.data['fig_idx']['idx']` as the key fixed the issue.